tests/pfquickcheck/pflang.lua

```diff
@@ -0,0 +1,219 @@
+-- initial lowercase letter => aux helper
+
+module(..., package.seeall)
+local choose = require("pf.utils").choose
+local set = require("pf.utils").set
+
+local function Empty() return { "" } end
+
+-- Given something like { 'host', '127.0.0.1' }, make it sometimes
+-- start with src or dst. This should only be called on expressions
+-- which can start with src or dst!
+local function srcdstify(expr)
+   local r = math.random()
+   if r < 1/3 then table.insert(expr, 1, "src")
+   elseif r < 2/3 then table.insert(expr, 1, "dst")
+   end -- else: leave it unchanged
+   return expr
+end
+
+local function IPProtocol()
+   return choose({ "icmp", "igmp", "igrp", "pim", "ah", "esp", "vrrp",
+                    "udp", "tcp", "sctp", "icmp6" })
+end
+
+-- These are parsed seperately from IPProtocol values in pflua; does it make
+-- sense to mainain this separation?
+local function MoreProtocols()
+   return choose({ "ip", "arp", "rarp", "ip6" })
+end
+
+local function ProtocolName()
+   return { choose({ IPProtocol, MoreProtocols })() }
+end
+
+local function portNumber()
+   return math.random(1, 2^16 - 1)
+end
+
+local function Port()
+   return { "port", portNumber() }
+end
+
+local function PortRange()
+   local port1, port2 = portNumber(), portNumber()
+   return { "portrange", port1 .. '-' .. port2 }
+end
+
+local function ProtocolWithPort()
+   protocol = choose({ "tcp", "udp" })
+   return { protocol, "port", portNumber() }
+end
+
+local function octet() return math.random(0, 255) end
+
+local function ipv4Addr()
+   return table.concat({ octet(), octet(), octet(), octet() }, '.')
+end
+
+local function Host()
+   return srcdstify({ 'host', ipv4Addr() })
+end
+
+local function netmask() return math.random(0, 32) end
+
+-- This function is overly conservative with zeroing octets.
+-- TODO: zero more precisely? It's trickier than it looks; tcpdump says
+-- 10.0.0.0/6 isn't ok ('non-network bits set'), but 10.0.0.0/7 is!
+function netspec()
+   local mask = netmask()
+   local o1, o2, o3, o4 = octet(), octet(), octet(), octet()
+   if mask < 32 then o4 = 0 end
+   if mask < 24 then o3 = 0 end
+   if mask < 16 then o2 = 0 end
+   if mask < 8 then o1 = 0 end
+   local addr = table.concat({ o1, o2, o3, o4 }, '.')
+   return addr .. '/' .. mask
+end
+
+function Net()
+   return srcdstify({ 'net', netspec() })
+end
+
+local function uint8() return math.random(0, 2^8-1) end
+
+local function uint16() return math.random(0, 2^16-1) end
+
+-- Boundary numbers are often particularly interesting; test them often
+local function uint32()
+   if math.random() < 0.2
+      then return math.random(0, 2^32 - 1)
+   else
+      return choose({ 0, 1, 2^31-1, 2^31, 2^32-1 })
+   end
+end
+
+-- ^ intentionally omitted; 'len < 1 ^ 1' is not valid pflang
+local function binaryMathOp()
+   return choose({ '+', '-', '/', '*', '|', '&' })
+end
+
+local function shiftOp() return choose({ '<<', '>>' }) end
```

Comment:
note that in pflang it seems to be perfectly valid to do 1>>33.  exercising this case will probably test undefined behavior in pflang though as the common C VMs don't check to ensure that the >> or << will be valid.